Chapter-3-Exercises/jjhampton-ch3-bean-counting.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+/* -------------------------------
+Write a function countBs that takes a string as its only argument and returns a number that indicates how many uppercase “B” characters are in the string.
+
+Next, write a function called countChar that behaves like countBs, except it takes a second argument that indicates the character that is to be counted (rather than counting only uppercase “B” characters). Rewrite countBs to make use of this new function.
+-------------------------------*/
+
+function countBs(stringToCheck, charToCount) {
+    const stringLength = stringToCheck.length; // cache string length to prevent repeated lookups during each for loop iteration
+    let counter = 0;
+    for (let i = 0; i < stringLength; i++) {
+        if (stringToCheck.charAt(i) === charToCount) 
+            counter++
+    }
+    return counter;
+}
+
```

Comment:
Like your example. I haven't got into the habit of using use strict on JS, but I need to. It saved me a lot of headache on a couple of VBA projects I did. So, why do you use const for stringLength? Is there a performance advantage versus having the loop use .length every time?